fcrepo-auth-webac/src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java

```diff
@@ -313,12 +321,26 @@ private FedoraResource getBaseVersion(final FedoraResource resource) {
     }
 
     /**
-     *  Given a FedoraResource, return a list of agents.
+     * Given a FedoraResource, return a list of agents.
      */
     private static Stream<String> getAgentMembers(final IdentifierConverter<Resource, FedoraResource> translator,
-            final FedoraResource resource) {
-        return resource.getTriples(translator, PROPERTIES).filter(memberTestFromTypes.apply(resource.getTypes()))
-            .map(Triple::getObject).flatMap(WebACRolesProvider::nodeToStringStream);
+                                                  final FedoraResource resource, final String hashPortion) {
+
+        //resolve list of triples, accounting for hash-uris.
+        final List<Triple> triples = resource.getTriples(translator, PROPERTIES).filter(
+            triple -> hashPortion == null || triple.getSubject().getURI().endsWith(hashPortion)).collect(toList());
+        //determine if there is a rdf:type vcard:Group
+        final boolean hasVcardGroup = triples.stream().filter(
+            triple -> triple.predicateMatches(RDF_TYPE_NODE) &&
+                      triple.getObject().getURI().equals(VCARD_GROUP_VALUE)).count() > 0;
```

Comment:
It is probably worthwhile to create and use constant nodes for VCARD_GROUP_VALUE and VCARD_MEMEBER_VALUE, like you did for rdf:type:
```
    private static final org.apache.jena.graph.Node VCARD_GROUP_NODE = createURI(VCARD_GROUP_VALUE);
    private static final org.apache.jena.graph.Node VCARD_MEMBER_NODE = createURI(VCARD_MEMBER_VALUE);
```